### PR TITLE
implement for hgraph_range

### PR DESCRIPTION
**Pull request by LHT129**

### CODE REVIEW

File: src/index/hgraph_index.cpp [hunk 1]
```diff
@@ -134,26 +134,26 @@ HGraphIndex::knn_search(const DatasetPtr& query,
         // check query vector
         CHECK_ARGUMENT(query->GetNumElements() == 1, "query dataset should contain 1 vector only");
 
-        auto ep = this->entry_point_id_;
+        InnerSearchParam search_param;
+        search_param.ep_ = this->entry_point_id_;
+        search_param.ef_ = 1;
```
Comment (inabao) [inline src/index/hgraph_index.cpp:150]: It is best to set the default value of ef to k.

File: src/index/hgraph_index.cpp [hunk 2]
```diff
@@ -662,27 +744,32 @@ HGraphIndex::calc_distance_by_id(const float* vector, int64_t id) const {
 }
 void
 HGraphIndex::add_one_point(const float* data, int level, InnerIdType inner_id) {
-    auto ep = this->entry_point_id_;
     MaxHeap result(allocator_);
+
+    InnerSearchParam param{
+        .ep_ = this->entry_point_id_,
```
Comment (inabao) [inline src/index/hgraph_index.cpp:761]: public fields do not need to end with `_`

Comment (LHT129) [reply src/index/hgraph_index.cpp:761]: rename all public member soon

File: src/index/hgraph_index.cpp (hunk 1 above)
Comment (LHT129) [reply src/index/hgraph_index.cpp:150]: 1 for high level graph

**Review by ShawnShawnYou — APPROVED** [review]

**Pull request merged**